src/components/Menu.vue

```diff
@@ -104,16 +116,22 @@ export default defineComponent({
       eComStore: 'user/getCurrentEComStore',
       instanceUrl: 'user/getInstanceUrl',
       userProfile: 'user/getUserProfile',
-      currentShopifyConfig: 'user/getCurrentShopifyConfig'
+      currentShopifyConfig: 'user/getCurrentShopifyConfig',
+      currentEComStore: 'user/getCurrentEComStore',
+      shopifyConfigs: 'user/getShopifyConfigs',
     })
   },
   methods: {
     async setEComStore(event: CustomEvent) {
       if(this.userProfile) {
         await this.store.dispatch('user/setEcomStore', { 'productStoreId': event.detail.value })
-        emitter.emit("productStoreChanged")
+        emitter.emit("productStoreOrConfigChanged")
       }
     },
+    async setShopifyConfig(event: CustomEvent){
+      await this.store.dispatch('user/setCurrentShopifyConfig', { 'shopifyConfigId': event.detail.value });
```

Comment:
In `setEComStore` we are checking for `userProfile` but in this method  we are not, any specific reason?

Reply:
The code is referenced from the Settings page. There, we are doing the same. 

I think the reason for using it on the Settings page is that the config setting is dependent on the ecomStores, hence if we check for the user profile before setting the ecomStore, we do not need to check the same for setting the configs.

Now, coming to our case, the Menu footer, on second thought, I think we do not need to check the same here, for any of the cases, as `this.userProfile` checks if the user is logged in or not and on tracing the code, only on successful login we are able to use the app and route guarding is there too.

So, I do not think it is required in any of the cases.

Reply:
As discussed, sir, the check is not needed here since we only need to handle this case when logging out from the settings page.